### ISSUE TITLE
Improve formatting on error message display

### ISSUE THREAD

**Issue by michaelmesser** [meraymond2/idris-vscode#23]
Error messages have too many new lines. Also VSCode already shows the location so it is not necessary to include the location in the message.

Current:
```
While
processing
right
hand
side
of imap. Can't
find
an
implementation
for FromString (IFunc (IVect n a) (IVect n b)).

/Users/michaelmesser/Code/Idris2FRP/src/Incremental.idr:76:14--76:17
    |
 76 |     imap f = "q"
    |              ^^^

```
Ideal:
```
While processing right hand side of imap.
Can't find an implementation for FromString (IFunc (IVect n a) (IVect n b)).
```
or
```
Can't find an implementation for FromString (IFunc (IVect n a) (IVect n b)).
```

**Comment by meraymond2** [comment]
Hmm, so at the moment, it transparently passes through the message that the IDE process returns, but I agree that some of that information is redundant within the editor. So given a warning like
```
(:warning ("Example.idr" (11 5) (11 8) "While processing right hand side of x at Example.idr:11:1--13:1:
When unifying String and Nat
Mismatch between:
	String
and
	Nat
at:
11	x = f 8
	    ^^^
" ()) 33)
```
I could probably strip away `While...:\n` and `at:...`, though I suspect it would be quite brittle. I could make it opt-in. I'll also have a quick look at the Idris2 code, to see if that preamble is defined in a single location, but I suspect that it's unique to each error.

Having said that, I've never seen anything like that first error with new lines after each word. Is that really what it looks like, or did it get mangled when copying? I get something more like
```
While processing right hand side of imap at Example.idr:12:1--14:1:
Can't find an implementation for FromString (Vect n a -> Vect n b) at:
12	imap = \"q\"
	       ^^^
```

**Comment by michaelmesser** [comment]
That is really what it looks like with`Idris 2, version 0.2.1-043faf8ba`. Idris 0.2.1 has some bugs that I fixed on GitHub so I built from source.

**Comment by michaelmesser** [comment]
Is there an easy way inspect the communication between idris and this? I want to see if idris is adding the new lines or if they are added afterwards.

**Comment by meraymond2** [comment]
Yes there is, set `debug` to true [in this line](https://github.com/meraymond2/idris-vscode/blob/2920704f2406cacab16a48b3156eafcaea3bd8f9/src/state.ts#L84), and it will print out all of the communication to and from the Idris process. If you've not run the extension locally before, it's just `npm install`, `npm run compile` and then the Start Debugging in VSCode, which is F5 on Linux.

I had a quick look at this yesterday, with Idris 2 built from the master branch, and couldn't reproduce the excessive new line bug. I might still take a stab at trimming the error messages, but it's slightly complicated by the fact that the format of the error messages have changed slightly since 0.2.1 was released.

**Comment by michaelmesser** [comment]
Thank you. That worked. I see the extra newlines in the debug console. I will try to figure out why Idris is generating them.

**Comment by michaelmesser** [comment]
As far as I can Idris 2 tries to get the width of the terminal it is printing to and it automatically wraps the text. I believe that is causing the issue.

**Comment by michaelmesser** [comment]
```
const procArgs = idris2Mode ? ["--ide-mode", "--find-ipkg", "--console-width", "0"] : ["--ide-mode"]
```

fixes it for me. Is this a bug in Idris 2 or should this package pass that option?

**Comment by michaelmesser** [comment]
It is weird that you don't experience this problem. I'm on macOS. Maybe it is OS dependant?

**Comment by michaelmesser** [comment]
I believe it is an Idris 2 bug. I was able to build a version of Idris 2 that does not have the issue by modifying two lines. I will make a PR to Idris 2 tomorrow.  Idris 2 gets the terminal width and it is 0 so it puts every word on a new line. I just had to add if the terminal width is zero then make it infinity. Idris does correctly interprets the --console-width argument when it is 0, but it does not interpret the actual terminal width being zero correctly.

**Comment by meraymond2** [comment]
Cool, thanks for investigating that :+1: . I am happy to have the extension pass in the 0 console width explicitly until it can get fixed in Idris 2. It doesn't have a noticeable change on my system.

It doesn't surprise me that it's using the terminal width, there's [an open issue](https://github.com/idris-lang/Idris2/issues/768) to improve the IDE protocol, because at the moment, it's mostly the same code used by the REPL. I think the IDE mode probably shouldn't have any concept of width. 

I'd like to make a release today, because there are a few pending fixes. The shorter error messages won't be ready in time, but can go into the next one. I don't think it's a difficult change, but it will need a lot of testing.

**Comment by michaelmesser** [comment]
Do you think Idris 2 should adopt Language Server Protocol instead of making its own protocol? I suggested that in the issue you mentioned, but I'm not sure it is a good idea or not yet.

**Comment by meraymond2** [comment]
Yes, I think it would be a good idea. I read through the other thread, and it sounds like there's broader enthusiasm for it, which I think is important for adoption. And I think people will be more willing to download an additional binary if it's an official language server.

I'll keep an eye on its progress :+1: 

(I also haven't forgotten about the error messages in the meantime, I just haven't had much energy for coding over the holidays, but I have started poking at this again.)

**Comment by meraymond2** [comment]
The shorter error messages have been released in 0.0.8.
